### PR TITLE
CONTRIBUTING: initial documentation

### PR DESCRIPTION
**Pull request by pamolloy**
This list was based on an internal checklist, but has been modified to be more generally useful

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+When contributing to Precision Converters Firmware please consider the
+following checklist:
+
+- [ ] Copyright header has been added to source and header files
+- [ ] Artistic Style (`astyle`) has been run to lint new code
```
Comment (pamolloy) [inline CONTRIBUTING.md:5]: I may add an link to the `.gitlab-ci.yml` or simply add an example like:

```bash
$ astyle --options=libraries/no-OS/ci/astyle_config projects/foobar/app/*.c
```

Comment (mphalke) [reply CONTRIBUTING.md:5]: I think example would be useful if easy to add here

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- [ ] Add EEPROM validation code
+- [ ] Add all context attributes (e.g. `fw_version`, `hw_carrier`,
+      `hw_mezzanine`, `hw_name`, `hw_mezzanine_status`)
+- [ ] IIO attributes match the corresponding Linux driver
+- [ ] Create a `readme.md` in the project directory explaining how to use the
+      project
+
+- [ ] Validate EEPROM detection and context attributes creation using IIO
+      clients (e.g. ACE, IIO Oscilloscope)
+- [ ] Capture data using a client
```
Comment (mphalke) [inline CONTRIBUTING.md:15]: nit: Capture and verify ADC data using an IIO client

**Comment by pamolloy** [comment]
@mphalke I just signed the CLA and rebased on `main` so I think it should be possible to merge this now

**Comment by mphalke** [comment]
> @mphalke I just signed the CLA and rebased on `main` so I think it should be possible to merge this now

@pamolloy I had given some comments on Sept 14, but looks like those were not reflected onto PR. Please see them and once addressed we are good to merge it

**Comment by pamolloy** [comment]
@mphalke this should be ready to be merged

**Review by mphalke — APPROVED** [review]

**Pull request merged**